crates/services/executor/src/executor.rs

```diff
@@ -322,11 +326,12 @@ where
             ExecutionTypes::DryRun(component) => {
```

Comment:
![image](https://github.com/FuelLabs/fuel-core/assets/11023047/e94bd3ae-2d6c-4a7c-9e10-452b39e9a8c0)
I thought you meant here.

Reply:
`validate_inner` it is the place where we want to keep this code=) The `execute_inner` is the place from which we want to remove it=)

Reply:
Okay. That makes sense. I wasn't even thinking about this side.

For `validate`, I moved it to
```
Self::check_new_block_id(partial_block, block, &data)?;
```
in `validate_block`. Do you see a problem with that?

Reply:
It can live in a separate function, but I would expect a function like `compare_blocks(prodecer_block, validator_block)` or something like that. Not only comparison of the `BlockId`=)